src/NuGet.Clients/NuGet.VisualStudio.Common/NuGetFeedbackDiagnosticFileProvider.cs

```diff
@@ -1,24 +1,110 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
 using Microsoft.Internal.VisualStudio.Shell.Embeddable.Feedback;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Common;
 using NuGet.PackageManagement;
+using NuGet.VisualStudio.Telemetry;
+
+#nullable enable
 
 namespace NuGet.VisualStudio.Common
 {
     [Export(typeof(IFeedbackDiagnosticFileProvider))]
     public class NuGetFeedbackDiagnosticFileProvider : IFeedbackDiagnosticFileProvider
     {
+        // Don't crash VS feedback if imports fail. Maybe broken MEF is exactly what the customer is trying to report!
+        [Import(AllowDefault = true)]
+        public ISolutionManager? SolutionManager { get; set; }
+
         public IReadOnlyCollection<string> GetFiles()
         {
-            // todo: https://github.com/NuGet/Home/issues/8605
+            string filePath = GetFilePath();
+
+            // See comments on IFeedbackDiagnosticFileProvider.GetFiles
+            ThreadHelper.JoinableTaskFactory.RunAsync(() => WriteToZip(filePath))
+                .PostOnFailure(nameof(NuGetFeedbackDiagnosticFileProvider), nameof(WriteToZip));
 
             return new List<string>()
             {
-                DependencyGraphRestoreUtility.GetDefaultDGSpecFileName()
+                filePath
             };
         }
+
+        public async Task WriteToZip(string filePath)
+        {
+            var telemetry = new TelemetryEvent("feedback");
+            telemetry["IsDebuggerAttached"] = Debugger.IsAttached;
+            bool successful = false;
+            Stopwatch sw = Stopwatch.StartNew();
+            try
+            {
+
+                using (var fileStream = File.Open(filePath, FileMode.Create, FileAccess.Write, FileShare.None))
+                using (var zip = new ZipArchive(fileStream, ZipArchiveMode.Create))
+                {
+                    await AddDgSpecAsync(zip);
+                }
+
+                successful = true;
+            }
+            finally
+            {
+                sw.Stop();
+                telemetry["successful"] = successful;
+                telemetry["duration_ms"] = sw.Elapsed.TotalMilliseconds;
+                TelemetryActivity.EmitTelemetryEvent(telemetry);
+            }
+        }
+
+        private string GetFilePath()
+        {
+            string fileNamePrefix = "nuget." + DateTime.UtcNow.ToString("yyyy-MM-dd.HH-mm-ss");
```

Comment:
Should we add `milloseconds` or simply add short `Guid` at the end to reduce collision possibility?
From code it looks like file storage is shared between many VSFeedbacks and I'm not sure about design behind it, in case it allows multithreading then 2 threads may return same `nuget.2022-07-26.21-47-18.zip` value since file is not created yet that exact `2022-07-26.21-47-18` moment (1 sec internal). But later when file is saved 1 thread may overwrite the other one's data since they got same file name. Adding Guid at the end will prevent it.

Reply:
The code immediately below the line you commented on deals with collisions.  The benefit of using the timestamp is that when we're on support and have multiple nuget.*.zip files in our downloads folder, the timestamp gives more information than a random guid (oh, I have 5 in my folder, which one am I currently trying to investigate? It's the one reported 3 days ago, oh look, it's easy to tell which zip is the relevant one).

Reply:
Sure, having timestamp is handy my proposal was `nuget.yyyy-MM-dd.HH-mm-ss-ShortGuid.zip` (`nuget.2022-07-26.21-47-18-404c-bbb4-671736b26c41.zip`). There're several kinds of short guid: https://www.codeproject.com/Tips/1236704/Reducing-the-string-Length-of-a-Guid or simply take half of generated GUID.
The code immediately below may return same value if 2 requests come in same second, then later unknowingly overwrite 1 another.

Reply:
I don't think that customers clicking the "report a problem" button more than once a second is going to happen very often, so I don't think it's worth trying to optimize for.